dvl-a50/dvl.py

```diff
@@ -382,7 +382,11 @@ def handle_velocity(self, data: Dict[str, Any]) -> None:
             self.mav.send_vision(position_delta, attitude_delta, dt=data["time"] * 1e3, confidence=confidence)
         elif self.should_send == MessageType.SPEED_ESTIMATE:
             velocity = [vx, vy, vz] if self.current_orientation == DVL_DOWN else [vz, vy, -vx]  # DVL_FORWARD
```

Comment:
Does the covariance matrix also need to be rotated?

Reply:
The DVL directionality isn't exposed in the frontend at the moment (it's just assumed to be down), but that is a good point - I expect the covariance matrix in the `DVL_FORWARD` case will need the same kind of reordering applied to it as the velocities, along with the negation of the x-related values (to match the reversal of that motion axis in the data).

Reply:
Added, including a top level note in the PR description.